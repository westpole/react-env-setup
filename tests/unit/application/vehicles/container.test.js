import toJson from 'enzyme-to-json';

import { mockWithStore } from '../../../tools/mock-component.jsx';
import Container from '../../../../src/scripts/application/vehicles/container';
import * as actions from '../../../../src/scripts/store/actions/core';

describe('Application/Vehicles/Container', () => {
  it('should match saved snapshot', () => {
    const wrapper = mockWithStore({
      type: 'shallow',
      scope: Container,
    });

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should connect Container to the Store', () => {
    const wrapper = mockWithStore({
      type: 'shallow',
      scope: Container,
    });

    expect(wrapper.instance().store).toBeDefined();
  });

  it('should dispatch an event to fetch vehicles data', () => {
    const spy = jest.spyOn(actions, 'fetchData');

    const wrapper = mockWithStore({
      type: 'shallow',
      scope: Container,
    });

    wrapper.props().fetchVehicles();

    expect(spy.mock.calls.length).toEqual(1);
  });
});
