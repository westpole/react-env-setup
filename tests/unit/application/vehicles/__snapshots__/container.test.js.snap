// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application/Vehicles/Container should match saved snapshot 1`] = `
<VehiclesComponent
  fetchVehicles={[Function]}
/>
`;
